### DISCUSSION TITLE
Synchronization Context - C#

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#120] [General]
Hi,

In my opinion, the following topic in this book needs more explanation and examples.
Synchronization Context - Chapter 11, Page 294.

Best

**Comment by christiannagel** [comment]
As described in this section, this is not relevant for console applications. The `SynchronizationContext` is very important for Windows applications. Just a few pages later (page 302, **Async with Windows Apps**) the issues with Windows applications are described - which should give some understanding of how the synchronization context helps with Windows applications.

Does this help?

**Comment by christiannagel** [comment]
I created an article and added additional source code to make it clearer what's given by the SynchronizationContext.
The article will be released next Tuesday on my blog https://csharp.christiannagel.com

Do you have any specific questions on this topic that I might add?

**Reply by ShervanN** [reply]
Thank you

**Comment by christiannagel** [comment]
The article is online: https://csharp.christiannagel.com/2022/09/06/whats-the-synchronizationcontext-used-for/

**Reply by ShervanN** [reply]
I appreciate it.
